src/lib/Lib/CMakeExecution.cpp

```diff
@@ -0,0 +1,164 @@
+    int const result = llvm::sys::ExecuteAndWait(*path, args, std::nullopt, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+    return *path;
+}
+
+Expected<std::string>
+executeCmakeHelp(llvm::StringRef cmakePath)
+{
+    llvm::SmallString<128> outputPath;
+    MRDOCS_CHECK(!llvm::sys::fs::createTemporaryFile("cmake-help", "txt", outputPath), 
+        "Failed to create temporary file");
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), outputPath.str(), llvm::StringRef()};
+    std::vector<llvm::StringRef> const args = {cmakePath, "--help"};
+    llvm::ArrayRef<llvm::StringRef> emptyEnv;
+    int const result = llvm::sys::ExecuteAndWait(cmakePath, args, emptyEnv, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+
+    auto const bufferOrError = llvm::MemoryBuffer::getFile(outputPath);
+    MRDOCS_CHECK(bufferOrError, "Failed to read CMake help output");
+
+    return bufferOrError.get()->getBuffer().str();
+}
+
+Expected<std::string>
+getCmakeDefaultGenerator(llvm::StringRef cmakePath) 
+{
+    MRDOCS_TRY(auto const cmakeHelp, executeCmakeHelp(cmakePath));
+
+    std::istringstream stream(cmakeHelp);
+    std::string line;
+    std::string defaultGenerator;
+
+    while (std::getline(stream, line)) {
+        if (line[0] == '*' && line[1] == ' ') {
+            size_t const start = 2;
+            size_t const end = line.find("=", start);
+            if (end == std::string::npos) {
+                continue;
+            }
+            return line.substr(start, end - start);
+        }
+    }
+    return Unexpected(Error("Default CMake generator not found"));
+}
+
+Expected<bool>
+cmakeDefaultGeneratorIsVisualStudio(llvm::StringRef cmakePath) 
+{
+    MRDOCS_TRY(auto const defaultGenerator, getCmakeDefaultGenerator(cmakePath));
+    return defaultGenerator.starts_with("Visual Studio");
+}
+
+std::vector<std::string>
+parseCmakeArgs(std::string const& cmakeArgsStr) 
+{
+    std::vector<std::string> parsedArgs;
+    std::string arg;
+    bool inQuotes = false;
+
+    for (char ch : cmakeArgsStr) 
+    {
+        if (ch == '"') 
+        {
+            inQuotes = !inQuotes;
+        } 
+        else if (std::isspace(ch) && !inQuotes) 
+        {
+            if ( ! arg.empty()) 
+            {
+                if (arg[0] == '-' && arg.size() == 2) 
+                    continue;
+                parsedArgs.push_back(arg);
+                arg.clear();                
+            }
+        } 
+        else 
+        {
+            arg += ch;
+        }
+    }
+
+    if ( ! arg.empty()) 
+    {
+        parsedArgs.push_back(arg);
+    }
+
+    return parsedArgs;
+}
+
+} // anonymous namespace
+
+Expected<std::string>
+executeCmakeExportCompileCommands(llvm::StringRef projectPath, llvm::StringRef cmakeArgs) 
+{
+    MRDOCS_TRY(auto const cmakePath, getCmakePath());
+    MRDOCS_CHECK(llvm::sys::fs::exists(projectPath), "CMakeLists.txt not found");
```

Comment:
This check should come before the call to getCmakePath

Reply:
The error message could be only that projectPath doesn't exist. It didn't check for CMakeLists.txt.

Reply:
The executeCmakeExportCompileCommands javadoc says projectPath can be the source path or CMakeLists.txt. This function doesn't handle both cases. It handles only the source path case (without checking if CMakeLists.txt exists).